Vagrantfile

```diff
@@ -21,83 +21,276 @@ Vagrant.configure("2") do |config|
+        'UPGRADE_PACKAGES': ENV['UPGRADE_PACKAGES'],
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        dnf -y upgrade ${UPGRADE_PACKAGES}
+    SHELL
+  end
 
-    # install runc (or crun) and other components
-    ./script/setup/install-runc
-    ./script/setup/install-cni
-    ./script/setup/install-critools
+  # To re-run, installing CNI from RPM:
+  #   INSTALL_PACKAGES="containernetworking-plugins" vagrant up --provision-with=install-packages
+  #
+  config.vm.provision "install-packages", type: "shell", run: "once" do |sh|
+    sh.upload_path = "/tmp/vagrant-install-packages"
+    sh.env = {
+        'INSTALL_PACKAGES': ENV['INSTALL_PACKAGES'],
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        dnf -y install \
+            container-selinux \
+            curl \
+            gcc \
+            git \
+            iptables \
+            less \
+            libseccomp-devel \
+            libselinux-devel \
+            lsof \
+            make \
+            mercurial \
+            selinux-policy \
+            selinux-policy-devel \
+            selinux-policy-targeted \
+            socat \
+            subversion \
+            vim \
+            wget \
+            ${INSTALL_PACKAGES}
+    SHELL
+  end
 
-    # install containerd
-    make BUILDTAGS="seccomp selinux no_aufs no_btrfs no_devmapper no_zfs" binaries install
+  # To re-run, installing a different version of go:
+  #   GO_VERSION="1.14.6" vagrant up --provision-with=install-golang
+  #
+  config.vm.provision "install-golang", type: "shell", run: "once" do |sh|
+    sh.upload_path = "/tmp/vagrant-install-golang"
+    sh.env = {
+        'GO_VERSION': ENV['GO_VERSION'] || "1.13.14",
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        _GO_DIR="/usr/lib/go${GO_VERSION}"
+        _GO_TMP=$(mktemp -p /tmp -d go.XXXX)
+        curl -fsSL "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" | tar -xz --strip=1 -C "${_GO_TMP}"
+        rm -fr "${_GO_DIR}"
+        mv -f "${_GO_TMP}" "${_GO_DIR}"
+        chmod -v og+rx "${_GO_DIR}"
+        rm -fv /usr/bin/{go,gofmt}
+        ln -fnsv "${_GO_DIR}/bin/go" "${_GO_DIR}/bin/gofmt" /usr/bin/
+        go env
+    SHELL
+  end
 
-    # FIXME: enable SELinux
-    setenforce 0
-    umount /sys/fs/selinux
+  config.vm.provision "prepare-containerd-source", type: "shell", run: "once" do |sh|
+    sh.upload_path = "/tmp/vagrant-prepare-containerd-source"
+    sh.env = {
+        'GOPATH': '/usr',
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        mkdir -p /etc/containerd ${GOPATH}/src/github.com/containerd
+        ln -fnsv /vagrant ${GOPATH}/src/github.com/containerd/containerd
+    SHELL
+  end
 
-    # create the daemon config
-    mkdir -p /etc/containerd
-    cat > /etc/containerd/config.toml <<EOF
+  # Disabled by default. To run:
```

Comment:
This comment is confusing as SELinux is enabled by default. "Disabled by default" -> "Not called by default" ?

Reply:
I see your point. It is a comment for the provisioner that by default is not enabled (`run: "never"`) but can be on the Vagrant CLI. SELinux itself is, of course, enabled and enforcing by default on Fedora/CentOS (and one assumes RHEL) but the `selinux-permissive` provisioner above is enabled which means that the VM will be in SELinux permissive mode when spun up with the default Vagrant provisioners.

Reply:
Just noticed that I hadn't pushed up a rev that put selinux-permissive above this.